src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.stream.IntStream;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_SIZE_ARRAY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE_ARRAY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        array[size] = (T) value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        grow();
+        if (index > size || index < 0 || array.length < index) {
+            throw new ArrayListIndexOutOfBoundsException("such an index is unacceptable");
+        }
```

Comment:
```suggestion
        if (index > size || index < 0) {
            throw new ArrayListIndexOutOfBoundsException("Such an index "+index+" is invalid for size " + size);
        }
```

lets make exception messages more informative